python/semantic_kernel/functions/kernel_function.py

```diff
@@ -0,0 +1,488 @@
+                if function_config.has_chat_prompt:
+                    messages = await function_config.prompt_template.render_messages(kernel, arguments)
+                    async for partial_content in client.complete_chat_stream(
+                        messages=messages, settings=request_settings
+                    ):
+                        yield partial_content
+                else:
+                    prompt = await function_config.prompt_template.render(kernel, arguments)
+                    async for partial_content in client.complete_stream(prompt, request_settings):
+                        yield partial_content
+
+            except Exception as e:
+                logger.error(f"Error occurred while invoking function {function.name}: {e}")
+                raise e
+
+        semantic_function_params = [
+            KernelParameterMetadata(
+                name="function",
+                description="The function to execute",
+                default_value=None,
+                type="KernelFunctionMetadata",
+                required=True,
+            ),
+            KernelParameterMetadata(
+                name="kernel",
+                description="The kernel",
+                default_value=None,
+                type="Kernel",
+                required=True,
+            ),
+            KernelParameterMetadata(
+                name="client",
+                description="The AI service client",
+                default_value=None,
+                type="AIServiceClientBase",
+                required=True,
+            ),
+            KernelParameterMetadata(
+                name="request_settings",
+                description="The request settings",
+                default_value=None,
+                type="PromptExecutionSettings",
+                required=True,
+            ),
+            KernelParameterMetadata(
+                name="arguments",
+                description="The kernel arguments",
+                default_value=None,
+                type="KernelArguments",
+                required=True,
+            ),
+        ]
+        semantic_function_params.extend(function_config.prompt_template.get_parameters())
+        return KernelFunction(
+            function_name=function_name,
+            plugin_name=plugin_name,
+            description=function_config.prompt_template_config.description,
+            function=_local_func,
+            parameters=semantic_function_params,
+            return_parameter=KernelParameterMetadata(
+                name="return",
+                description="The completion result",
+                default_value=None,
+                type="FunctionResult",
+                required=True,
+            ),
+            stream_function=_local_stream_func,
+            is_semantic=True,
+            chat_prompt_template=function_config.prompt_template if function_config.has_chat_prompt else None,
+        )
+
+    @property
+    def name(self) -> str:
```

Comment:
I think these properties should be gone now that we made kernel_function pydantic.